solr/core/src/java/org/apache/solr/util/circuitbreaker/CPUCircuitBreaker.java

```diff
@@ -110,6 +113,27 @@ public double getCpuUsageThreshold() {
   }
 
   protected double calculateLiveCPUUsage() {
-    return operatingSystemMXBean.getSystemLoadAverage();
+    Metric metric = this.core
+        .getCoreContainer()
+        .getMetricManager()
+        .registry("solr.jvm")
+        .getMetrics()
+        .get("os.systemCpuLoad");
```

Comment:
I dont quite understand this change -- so you are replacing the default CPU circuit breaker with a new implementation and moving the earlier implementation to a new class?

IMO, both the circuit breakers operate on CPU, using different metrics. Both circuit breakers should be named in a way which represents the fact that they operate on CPU and highlight which metric they depend on.

Reply:
The existing CPU CB is actually a load average CB. Load average is implented differently in different OSes and is not always purely CPU. Also, load average is no limited to 0-100 as documented.

The existing CB is renamed to an accurate name with accurate documentation.

A new CB is created that does what the original CB was documented to do, based on CPU usage instead of load average.

Reply:
The load average represents CPU usage and resources waiting for CPU - - so the job queue will contribute to the load average.

AFAIK, load average will always include CPU and CPU dependent tasks, which is a good representation of the load on the system.

Nevertheless, I am not a stickler for the naming as long as the documentation is correct

Reply:
The documentation and the linked Wikipedia page give more information on load average.

Load average is a "system" load average, not just CPU. From Wikipedia:

"An idle computer has a load number of 0 (the idle process isn't counted). Each process using or waiting for CPU (the ready queue or run queue) increments the load number by 1. Each process that terminates decrements it by 1. Most UNIX systems count only processes in the running (on CPU) or runnable (waiting for CPU) states. However, Linux also includes processes in uninterruptible sleep states (usually waiting for disk activity), which can lead to markedly different results if many processes remain blocked in I/O due to a busy or stalled I/O system."